tests/server/openDirectConnection.ts

```diff
@@ -218,3 +218,49 @@ test('direct connection transact awaits until onStoreDocument has finished, even
     resolve('done')
   })
 })
+
+test('creating a websocket connection after transact but before debounce interval doesnt create different docs', async t => {
+  let onStoreDocumentFinished = false
+  let disconnected = false
+
+  await new Promise(async resolve => {
```

Comment:
This Promise wrapper here looks unnecessary to me, as it's used inside an async function anyway and ultimately wraps another async function.

Reply:
Otherwise, this test case seems to capture the use case 👍

Reply:
Run the test against the unfixed `main` version and it failed, so it does indeed capture the error and the fix. Nice!